### PR TITLE
[AME-1281] Add a2_ validation to account id input and rename from advertiser ID to ad account ID 

### PR DESCRIPTION
**Pull request by elmiranda**
## 💸 TL;DR

Part of the advertiser id --> ad account id migration! Intent is to remove requirements that the id conforms to _only_ `t2_` ~and make sure it looks spick and span~ and allow them to be prefixed with `a2_`.

### CODE REVIEW

File: template.tpl [hunk 1]
```diff
@@ -41,19 +41,12 @@ ___TEMPLATE_PARAMETERS___
...
-          "t2_[a-z0-9]+"
-        ],
-        "type": "REGEX"
       }
     ],
-    "displayName": "Advertiser ID",
+    "displayName": "Ad Account ID",
     "simpleValueType": true,
     "name": "id",
     "type": "TEXT",
```
Comment (woodchuckgym) [inline template.tpl:49]: sn00bie question: is the trailing comma here OK?  That is, that's not valid JSON, but it is valid JavaScript.  (Was this file autogenerated, or edited by hand?)

**Review by woodchuckgym — APPROVED** [review]
TY!  One question about formatting.

**Review by dmaii — APPROVED** [review]
Assuming that this was tested in GTM and it's ok with the JSON, LGTM!

File: template.tpl (hunk 1 above)
Comment (elmiranda) [reply template.tpl:49]: Autogenerated!

Comment (elmiranda) [reply template.tpl:49]: Excellent question!

Comment (elmiranda) [reply template.tpl]: Good catch, I tried the replacement again and it should be fixed now. Bad copy pasta, sorry!

**Review by monicating — APPROVED** [review]

**Comment by elmiranda** [comment]
Hey folks, I'm retagging because I've updated (and rebased) this PR via https://github.com/reddit/reddit-gtm-template/pull/18/commits/37011455c86074ae4e9caa2225bd1e8d753b2542; this commit is basically the entire change set. I'll deploy once it's reapproved.

P.S. I told gym I would do this like 2 weeks ago but I got locked out of my Github account! Whoopie! We gucci now.

File: template.tpl [hunk 2]
```diff
@@ -44,16 +44,16 @@ ___TEMPLATE_PARAMETERS___
       },
       {
         "args": [
-          "t2_[a-z0-9]+"
+          "(t2_|a2_)[a-z0-9]+"
```
Comment (woodchuckgym) [inline template.tpl:47]: Were you able to test this in the tag manager?  I see the unit test -- which is super! -- but a test from the UI would be nice confirmation it works as intended.  If you haven't played with tagmanager.google.com before, I can show you.

File: template.tpl [hunk 3]
```diff
@@ -910,6 +910,30 @@ scenarios:
     // Call runCode to run the template's code.
     runCode(mockData);
 
+    // Verify that the tag finished successfully.
+    assertApi('gtmOnSuccess').wasCalled();
+- name: Test pixel init - set a2_ advertiser ID and integration type
```
Comment (woodchuckgym) [inline template.tpl:915]: TY!

**Review by woodchuckgym — CHANGES_REQUESTED** [review]
"Requesting changes" for a quick end-to-end test with the newly-committed changes for the business account IDs.

**Review by dmaii — APPROVED** [review]
Agree with Jim but LGTM otherwise

**Comment by elmiranda** [comment]
Tested this as a proper tag in the wild, on a test webpage! S/O to Austin who actually has a persistent setup for a GTM tag for exactly this reason. We tested both `t2_33rjacqk` and `a2_bo1kbw5ereg5 `, his account and the test account, respectively. I was able to confirm that the data landed successfully in BQ (no degradation from previous tests).

image

**Review by woodchuckgym — APPROVED** [review]
TY!

**Pull request merged**